CwdUtils.c

```diff
@@ -0,0 +1,158 @@
+#include "config.h" // IWYU pragma: keep
+
+#include "CwdUtils.h"
+
+#include <stdbool.h>
+#include <stdlib.h>
+#include <wchar.h>
+
+#include "XUtils.h"
+
+typedef struct ShortenCwdContext {
+   size_t maxLength;
+   size_t len;
+   wchar_t **parts;
+   size_t partsLen;
+   size_t *partLengths;
```

Comment:
The fields `partsLen` and `partLengths` have quite similar names which may cause confusion.

To mitigate this, using a naming scheme like `strSize`/`partCount`/`partPtr`/`partSize` might be somewhat more easy to disambiguate.

Reply:
Should be `partPtrs` and `partSizes` because the latter two are arrays.